sunkit_image/utils.py

```diff
@@ -0,0 +1,517 @@
+
+    Parameters
+    ----------
+    smap : sunpy.map.Map
+        A sunpy map.
+
+    radial_bin_edges : `~astropy.units.Quantity`
+        A two-dimensional array of bin edges of shape (2, nbins) where nbins is
+        the number of bins.
+
+    scale : None, `~astropy.units.Quantity`
+        A length scale against which radial distances are measured, expressed
+        in the map spatial units. For example, in AIA helioprojective
+        Cartesian maps a useful length scale is the solar radius and is
+        expressed in units of arcseconds.
+
+    summary : `~function`
+        A function that returns a summary statistic of the distribution of intensity,
+        at a given radius, for example `~numpy.std`.
+
+    summary_kwargs :`~dict`
+        Keywords applicable to the summary function.
+
+    Returns
+    -------
+    intensity summary : `~numpy.array`
+        A summary statistic of the radial intensity in the bins defined by the
+        bin edges.
+    """
+    if scale is None:
+        s = smap.rsun_obs
+    else:
+        s = scale
+
+    # Get the radial distance of every pixel from the center of the Sun.
+    map_r = find_pixel_radii(smap, scale=s).to(u.R_sun)
+
+    # Number of radial bins
+    nbins = radial_bin_edges.shape[1]
+
+    # Upper and lower edges
+    lower_edge = [map_r > radial_bin_edges[0, i].to(u.R_sun) for i in range(0, nbins)]
+    upper_edge = [map_r < radial_bin_edges[1, i].to(u.R_sun) for i in range(0, nbins)]
+
+    # Calculate the summary statistic in the radial bins.
+    return np.asarray(
+        [
+            summary(smap.data[lower_edge[i] * upper_edge[i]], **summary_kwargs)
+            for i in range(0, nbins)
+        ]
+    )
+
+
+# The functions below this are subroutines for the OCCULT 2.
+def bandpass_filter(image, nsm1=1, nsm2=3):
+    """
+    Applies a band pass filter to the image.
+
+    Parameters
+    ----------
+    image : `numpy.ndarray`
+        Image to be filtered.
+    nsm1 : `int`
+        Low pass filter boxcar smoothing constant.
+        Defaults to 1.
+    nsm2 : `int`
+        High pass filter boxcar smoothing constant.
+        The value of `nsm2` equal to `nsm1 + 1` gives the best enhancement.
+        Defaults to 3.
+
+    Returns
+    -------
+    `numpy.ndarray`
+        Bandpass filtered image.
+    """
+
+    if nsm1 >= nsm2:
+        raise ValueError("nsm1 should be less than nsm2")
+
+    if nsm1 <= 2:
+        return image - smooth(image, nsm2, "replace")
+
+    if nsm1 >= 3:
+        return smooth(image, nsm1, "replace") - smooth(image, nsm2, "replace")
+
+
+def smooth(image, width, nanopt="replace"): 
+    """
+    Python implementation of the IDL `smooth <https://www.harrisgeospatial.com/docs/smooth.html>`__.
+
+    Parameters
+    ----------
+    image : `numpy.ndarray`
+        Image to be filtered.
+    width : `int`
+        Width of the boxcar. The `width` should always be odd but if even value is given `width + 1` as the width of the boxcar.
```

Comment:
line is a bit too long

Reply:
same here, what should I remove from here

Reply:
You should just linebreak it into two lines.